cabal-install/src/Distribution/Client/CmdHaddockProject.hs

```diff
@@ -138,6 +139,13 @@ haddockProjectAction flags _extraArgs globalFlags = do
                        { configVerbosity = haddockProjectVerbosity flags }
                    }
 
+    unless (   fromFlagOrDefault False (haddockProjectHackage flags)
+            || fromFlagOrDefault False (haddockProjectLocal flags)
```

Comment:
I don't think I'm comfortable with the command doing most things but not quite everything without an additional flag. From a user perspective the most common use-case would be `--local` and I'd like to suggest it to be the default unless `--hackage` is passed.

Reply:
But then, there's no way to use `haddock` the way one would want, e.g. maybe one is not interested in creating the `index.html` because it's created in another way.

I am not sure how useful this is though...

Reply:
yeah I actually have never seen anyone able to generate the index outside of the "normal"(sic) workflow. Worst case, we can modularise it later, but I'd like to invest in usability first. :)

Reply:
+1 for having `--local` on by default unless `--hackage` is there.

Reply:
Done, please review.